### DISCUSSION TITLE
ConnectorValidator not working for TitledSkin

### DISCUSSION THREAD

**Discussion by amitjoy** [eckig/graph-editor#102] [Q&A] [answered]
I have added the following validator to the TitledSkin (as you proposed for 1-many and many-1 connections), but it doesn't work for TitledSkin whereas it works perfectly for DefaultSkin.

N.B: I did add the following in `io.github.eckig.grapheditor.demo.customskins.TitledSkinController.activate()`

```
graphEditor.setConnectorValidator(new DefaultConnectorValidator() {
            @Override
            public boolean validate(final GConnector source, final GConnector target) {

                if (source.getType() == null || target.getType() == null || source.getParent().equals(target.getParent())) {
                    return false;
                }

                final var sourceIsInput = DefaultConnectorTypes.isInput(source.getType());
                final var targetIsInput = DefaultConnectorTypes.isInput(target.getType());

                return sourceIsInput != targetIsInput;
            }
        });
```

**Comment by eckig** [comment, marked as answer]
Not the best place to change the validator, because of the order of listeners: The original demo validator is applied after that in `io.github.eckig.grapheditor.demo.GraphEditorDemoController#handleActiveSkinControllerChange`.

But with a quick `Platform.runLater()` hack the validator is applied correctly.

**Reply by amitjoy** [reply]
Thanks a lot for your continuous support 👍 It works perfectly now. As you proposed, I am going to add the validator to the graph init so that it gets applied without running it using `Platform#runLater`.
